src/app/components/SwapSubmitModal/index.jsx

```diff
@@ -137,21 +168,32 @@ export default compose(
         toastr.error(e.message || e)
         log.error(e)
       })
+    },
+    handleCheckBoxValidation: () => (checked) => {
+      if (!checked) {
+        return 'Please accept the Faast Terms in order to submit your swap.'
+      }
     }
   }),
   branch(
     ({ swundle }) => !swundle,
     renderNothing
   ),
-  withProps(({ swundle, requiresSigning, readyToSign, readyToSend, startedSigning, startedSending, handleSendTxs, handleSignTxs }) => {
+  withStateHandlers(
+    { timerExpired: false },
+    { 
+      handleTimerEnd: () => () => ({ timerExpired: true })
+    },
+  ),
+  withProps(({ swundle, requiresSigning, readyToSign, readyToSend, startedSigning, startedSending, handleSendTxs, handleSignTxs, formErrors }) => {
     let errorMessage = swundle.error
     const soonestPriceExpiry = min(swundle.swaps.map(swap => swap.rateLockedUntil))
     const secondsUntilPriceExpiry = (Date.parse(soonestPriceExpiry) - Date.now()) / 1000
     const currentSwap = swundle.swaps.find(({ txSigning, txSending, sendWallet }) =>
       txSigning || (txSending && sendWallet && !sendWallet.isSignTxSupported))
     const showSubmit = !requiresSigning || startedSigning // True if continue button triggers tx sending, false for signing
     const handleContinue = showSubmit ? handleSendTxs : handleSignTxs
```

Comment:
Remove handleContinue from onClick on the continue button and instead add a form onSubmit handler that calls handleSendTxs or handleSignTxs instead